src/utils/storage.jsx

```diff
@@ -0,0 +1,84 @@
+// Throttled storage helper to prevent excessive writes
+let storageTimeout = null;
+const STORAGE_DELAY = 1000; // Save every second
+
+export const saveTimerState = (state) => {
```

Comment:
You should be saving not only the timer state, but the timer queue as well. That way, when a user refreshes the page, the queue remains intact. When I refresh the page now, I lose my entire workout queue.